src/codecs/encoding/encoder.rs

```diff
@@ -322,4 +323,23 @@ mod tests {
         let sink = framed.into_inner();
         assert_eq!(sink, b"(foo)(bar)");
     }
+
+    #[tokio::test]
+    async fn test_encode_batch_newline() {
+        let encoder = Encoder::<Framer>::new(
+            Framer::NewlineDelimited(NewlineDelimitedEncoder::default()),
+            TextSerializerConfig::default().build().into(),
+        );
+        let source = futures::stream::iter(vec![
+            Event::Log(LogEvent::from("bar")),
+            Event::Log(LogEvent::from("baz")),
+            Event::Log(LogEvent::from("bat")),
+        ])
+        .map(Ok);
+        let sink: Vec<u8> = Vec::new();
+        let mut framed = FramedWrite::new(sink, encoder);
+        source.forward(&mut framed).await.unwrap();
+        let sink = framed.into_inner();
+        assert_eq!(sink, b"bar\nbaz\nbat\n");
```

Comment:
It looks like the expectation of a few test cases in `sinks/util/encoding` needs to be updated accordingly (seems we exercised this code path in unit tests already but the expectation was wrong.

Reply:
👍 updated the tests.